Packs/Code42/Integrations/Code42/Code42.py

```diff
@@ -15,509 +17,1179 @@
+    alert_filters.append_result(_get_severity_filter_value(severity), Severity.is_in)
+    alert_filters.append(AlertState.eq(AlertState.OPEN))
+    alert_filters.append_result(start_time, DateObserved.on_or_after)
+    alert_query = alert_filters.to_all_query()
+    return alert_query
+
+
+def _get_all_high_risk_employees_from_page(page, risk_tags):
+    res = []
+    # Note: page is a `Py42Response` and has no `get()` method.
+    employees = page["items"]
+    for employee in employees:
+        if not risk_tags:
+            res.append(employee)
+            continue
+
+        employee_tags = employee.get("riskFactors")
+        # If the employee risk tags contain all the given risk tags
+        if employee_tags and set(risk_tags) <= set(employee_tags):
+            res.append(employee)
+    return res
+
+
+def _try_convert_str_list_to_list(str_list):
+    if isinstance(str_list, str):
+        return str_list.split(",")
+    return str_list
 
 
 class Code42Client(BaseClient):
     """
-    Client will implement the service API, should not contain Demisto logic.
+    Client will implement the service API, should not contain Cortex XSOAR logic.
     Should do requests and return data
     """
 
     def __init__(self, sdk, base_url, auth, verify=True, proxy=False):
         super().__init__(base_url, verify=verify, proxy=proxy)
-        # Create the Code42 SDK instnace
-        self._sdk = sdk.create_using_local_account(base_url, auth[0], auth[1])
-        py42.settings.set_user_agent_suffix("Demisto")
-
-    def add_user_to_departing_employee(self, username, departure_epoch=None, note=None):
-        de = self._sdk.employee_case_management.departing_employee
-        try:
-            res = de.create_departing_employee(username, departure_epoch=departure_epoch, notes=note)
-        except Exception:
-            return None
-        return res.json().get('caseId')
-
-    def fetch_alerts(self, start_time, event_severity_filter=None):
-        alert_filter = []
-        # Create alert filter
-        if event_severity_filter:
-            alert_filter.append(Severity.is_in(list(map(lambda x: x.upper(), event_severity_filter))))
-        alert_filter.append(AlertState.eq(AlertState.OPEN))
-        alert_filter.append(DateObserved.on_or_after(start_time))
-        alert_query = AlertQuery(self._sdk.user_context.get_current_tenant_id(), *alert_filter)
-        alert_query.sort_direction = "asc"
-        alerts = self._sdk.security.alerts
-        try:
-            res = alerts.search_alerts(alert_query)
-        except Exception:
-            return None
-        return res.json().get('alerts')
+        # Allow sdk parameter for unit testing.
+        # Otherwise, lazily load the SDK so that the TEST Command can effectively check auth.
+        self._sdk = sdk
+        self._sdk_factory = (
+            lambda: py42.sdk.from_local_account(base_url, auth[0], auth[1])
+            if not self._sdk
+            else None
+        )
```

Comment:
Creating the sdk checks auth and the `test` command is supposed to do that, so I didn't want to check for auth right away so that the test command actually works. Do you have another recommended way?